src/test/java/com/ge/predix/metering/filter/NuregoIT.java

```diff
@@ -56,85 +89,237 @@
+            final  MockHttpServletRequest request, final ServletResponse response, final boolean acsFlag) throws Exception {
+    		System.out.println("==================================" + featureId + "==========================================");
+    		try {   
+	    	  	String guid = uaaGuid;
+	     	if (acsFlag) {
+	     		guid = serviceInstanceGuid;
+	     	}
+		    	String accessToken = getNuregoAuthToken();   	
+		    	System.out.println("accessToken::"+accessToken);	
+		    	String componentId= retrieveComponent(guid, accessToken);
+		    	if(componentId == null) {
+		    		throw new IllegalStateException("Component ID is null because Nurego mapping is not updated.");
+		    	}
+	
+		   	System.out.println("componentId::"+componentId);
+		    	
+		    this.meteringFilter.doFilter(request, response, new MockFilterChain());
+		    Thread.sleep(4000);
+		    this.meteringFilter.doFilter(request, response, new MockFilterChain());
+		    Thread.sleep(4000);
+	
+		    ComponentUsage usages = retrieveRawUsage(accessToken, componentId, acsFlag, featureId);
+		    Assert.assertEquals(usages.getCompCount(), 2);
+		    Assert.assertEquals(usages.getUsage(), 2.0);
+    		} catch(Exception e){
+    			e.printStackTrace();
+    		}
+    }
+   
+    @AfterTest
+    private void cleanup() {
+        cfClientTest.deleteServiceInstance(serviceInstanceGuid);
+        cfClientTest.deleteServiceInstance(uaaGuid);
+        System.out.println("Successfully deleted " + serviceInstanceGuid);
+        System.out.println("Successfully deleted " + uaaGuid);
+ 
+    }
+    private String getNuregoAuthToken() throws Exception{
+    	
+    		HashMap<String, Object> serviceRequest = new HashMap<String, Object>();
+		serviceRequest.put(Constants.USERNAME, this.nuregoUsername);
+		serviceRequest.put(Constants.PASSWORD, this.nuregoPassword);
+		serviceRequest.put(Constants.INSTANCE_ID, this.nuregoInstanceId);
 
-    private Entitlement getEntitlementByFeatureId(final String featureId, final String subscriptionId)
-            throws Exception {
-        List<Entitlement> entitlements = Entitlement.retrieve(subscriptionId).getData();
-        for (Entitlement entitlement : entitlements) {
-            if (entitlement.getFeatureId().equals(featureId)) {
-                return entitlement;
-            }
-        }
+		URI oauthURL = URI.create(this.nuregoOauthURL);
+		String response = this.nuregoTemplate.postForObject(oauthURL, serviceRequest, String.class);
 
-        return null;
+		@SuppressWarnings("unchecked")
+		Map<String, Object> responseMap = new ObjectMapper().readValue(response, Map.class);
+		return responseMap.get(Constants.ACCESS_TOKEN).toString();
     }
+    
+	private String retrieveComponent(String serviceInstanceGuid, String accessToken) throws Exception{
+
+		String retrieveComponentURL = this.nuregoURL + Constants.SERVICE_COMPONENT_URL+serviceInstanceGuid;
+		
+		HttpHeaders headers = new HttpHeaders();
```

Comment:
Headers can be moved to setup method as well and can be reused in the class, instead of creating them everytime.